src/OauthTokenFileStorage.php

```diff
@@ -263,7 +264,7 @@ private function getFromStorage(): array {
     $data = [];
     // Get the token data from the file store.
     if (file_exists($this->tokenFilePath) && ($raw_data = file_get_contents($this->tokenFilePath))) {
-      $data = unserialize(base64_decode($raw_data));
```

Comment:
I do not have access to the original security issue, but the token is coming from a trusted system  and if there is a MTM attack in place between Drupal and the Apigee instance, it is already pretty bad and this does not fix that problem. 
So my question is, does this fix really necessary?
https://www.php.net/manual/en/function.unserialize.php

json_decode() has a worse performance: https://grechin.org/2021/04/06/php-json-encode-vs-serialize-performance-comparison.html

Reply:
@mxr576 We evaluated that it was very highly unlikely that this was going to be an issue. It would need for someone to gain access to the servers to make this changes. As per the drupal security team it is not a bug and hence the direction to fix it. 

I do not see performance as an issue since we are probably caching the token after the first page load. 
@shishir-intelli to confirm the last statement.

Reply:
@giteshk Yes correct, just storing the token in `oauth.dat` file using Json encode/decode. It wont impact the performance.

Reply:
But my point is thaz the data stored in the file gets decoded every time when a new sdk connector service is being initiated for making API requests. This may not have a performance impact ofc, I havr just wanted call yout attention that it may requires profiling. 

So it is not going to be promoted as a security fix after all?

Reply:
Yep it is not promoted as a security fix. 

Thank you for bringing into notice the potential impact this could have.